public/maillog.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/*
+* to do :   - add pagination
+*	          - sort by mounth
+* 	        - remove old logs
+*/
+
+require_once('common.php');
+
+authentication_require_role('admin');
+
+$CONF = Config::getInstance()->getAll();
+$smarty = PFASmarty::getInstance();
+
+$PALANG = $CONF['__LANG'];
+
+$SESSID_USERNAME = authentication_get_username();
+if (authentication_has_role('global-admin')) {
+    $list_domains = list_domains();
+} else {
+    $list_domains = list_domains_for_admin($SESSID_USERNAME);
+}
+
+
+/*
+foreach ($list_domains as $domain){
+	echo $domain. '<br>';
+}
+*/
+
+$fDomain=$list_domains[0];
+
+if ($_SERVER['REQUEST_METHOD'] == "GET") {
+    if (isset($_GET['fDomain'])) {
+        $fDomain_aux = escape_string($_GET['fDomain']);
+        $flag_fDomain = 0;
+	//check if domain exists
+        if ((is_array($list_domains) and sizeof($list_domains) > 0)) {
+            foreach ($list_domains as $domain) {
+                if ($domain == $fDomain_aux) {
+                    $fDomain=$domain;
+                    $flag_fDomain=1;
+                    break;
+                }
+            }
+        }
+    
+        if ($flag_fDomain == 0 ) {
+            die('Unknown domain');
+        }
+	
+	if (strpos($fDomain, '/') !== false) { 
+                        die("Unknown path");
+                }
+
+    	//check if file exists
+   
+   	if (isset($_GET['get_log'])){
+		// remove path from filename
+		if (strpos($_GET['get_log'], '/') !== false) { 
+    			die("Unknown file");
+		}
+		
+	
+		$file = '../maillog/'.$fDomain.'/'.$_GET['get_log'];
+
```

Comment:
If you had .... 
if (!file_exists($file)) { die(.....) ; } 
here, you could un-indent the header/readfile block, which might make it easier to read.